eng/native/init-vs-env.cmd

```diff
@@ -4,10 +4,17 @@
 :: as an argument, it also initializes VC++ build environment and CMakePath.
 
 set "__VCBuildArch="
-if /i "%~1" == "x86"   (set __VCBuildArch=x86)
-if /i "%~1" == "x64"   (set __VCBuildArch=x86_amd64)
-if /i "%~1" == "arm64" (if /i "%PROCESSOR_ARCHITECTURE%" == "ARM64" (set __VCBuildArch=arm64) else (set __VCBuildArch=x86_arm64))
-if /i "%~1" == "wasm" (if /i "%PROCESSOR_ARCHITECTURE%" == "ARM64" (set __VCBuildArch=arm64) else (set __VCBuildArch=x86_amd64))
+if /i "%PROCESSOR_ARCHITECTURE%" == "ARM64" (
+    if /i "%~1" == "x64"   ( set __VCBuildArch=arm64_amd64 )
+    if /i "%~1" == "x86"   ( set __VCBuildArch=arm64_x86 )
+    if /i "%~1" == "arm64" ( set __VCBuildArch=arm64 )
+    if /i "%~1" == "wasm"  ( set __VCBuildArch=arm64 )
+) else (
+    if /i "%~1" == "x64"   ( set __VCBuildArch=amd64 )
+    if /i "%~1" == "x86"   ( set __VCBuildArch=amd64_x86 )
+    if /i "%~1" == "arm64" ( set __VCBuildArch=amd64_arm64 )
+    if /i "%~1" == "wasm"  ( set __VCBuildArch=amd64 )
```

Comment:
```suggestion
    set __VCBuildArch=amd64
    if /i "%~1" == "x86"   ( set __VCBuildArch=amd64_x86 )
    if /i "%~1" == "arm64" ( set __VCBuildArch=amd64_arm64 )
```

Reply:
Actually, that's a regression because the script relies on not setting `__VCBuildArch` if an argument hasn't been specified. I'll revert these two commits.